components/brave_vpn/resources/panel/components/main-panel/index.tsx

```diff
@@ -107,6 +107,14 @@ function MainPanel() {
   const connectionStatus = useSelector((state) => state.connectionStatus)
   const expired = useSelector((state) => state.expired)
   const regions = useSelector((state) => state.regions)
+  const [showSelectedRegionOnce, setShowSelectedRegionOnce] =
+    React.useState(window.location.pathname === '/select')
+
+  if (showSelectedRegionOnce) {
+    setShowSelectedRegionOnce(false)
```

Comment:
I don't think we should call `setShowSelectedRegionOnce` or `dispatch` in the render method - instead, we should wrap it in a `useEffect`

```ts
useEffect(() => {
  if (!showSelectedRegionOnce) return

  setShowSelectedRegionOnce(false)
  dispatch(Actions.toggleRegionSelector(true))
}, []) // empty deps array will only be called onMount
```

out of interest, do we even need this 

`if (showSelectedRegionOnce) return null`?

would this work?

```ts
useEffect(() => {
  if (window.location.pathname !== '/select') return

  dispatch(Actions.toggleRegionSelector(true))

Reply:
I used `useState` to avoid main page rendering before showing selection page. so early returned `null`.
With `useEffect()`, I can't avoid it.
but I think we should set `isSelectingRegion`'s initial state to `true` instead of handling here.
I'll try in that way.

Reply:
Pushed https://github.com/brave/brave-core/pull/26310/commits/8576ec5a8f86bb79a5e06469c9782fcf1b5af22e wdyt?

Reply:
looks perfect!